core/state/snapshot/snapshot.go

```diff
@@ -818,17 +818,21 @@ func (t *Tree) Verify(root common.Hash) error {
 // The lock of snapTree is assumed to be held already.
 func (t *Tree) disklayer() *diskLayer {
 	var snap snapshot
+	t.lock.RLock()
```

Comment:
The description said the snap tree lock is assumed to be held here. Please revert this change.

However, i do find in one place we forget to add the lock, please fix it there.

```go
// Release releases resources
func (t *Tree) Release() {
	t.lock.RLock()
	defer t.lock.RUnlock()

	if dl := t.disklayer(); dl != nil {
		dl.Release()
	}
}
```

Reply:
reverted!